### PR TITLE
Add GHA sample code to subdirs

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-external-examples/331)

### CODE REVIEW

**Review by jwnimmer-tri — COMMENTED** [review]
@nicolecheetham @BetsyMcPhail here is a sample for the next step on GHA hints in DEE.

We'll plan to close this PR once any Q&A is finished, y'all can create a new PR that repeats this idea for all examples.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-external-examples/331#-:-OA-j1P6-xZ3x0Fn8edx:b-w827ra)* status: all discussions resolved, platform LGTM missing

___
*[`.github/workflows/cmake_installed.yml` line 27 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-external-examples/331#-OA-i5ph2igOdYyy-V2X:-OA-i5ph2igOdYyy-V2Y:b-7ochbs) ([raw file](https://github.com/RobotLocomotion/drake-external-examples/blob/4192d150ec8f3ae71806011ea06c14145745afba/.github/workflows/cmake_installed.yml#L27)):*
> ```YAML
>           sudo rm -rf /Library/Frameworks/Python.framework/
>       - name: setup
>         working-directory: drake_cmake_installed
> ```

FYI By splitting out "working-directory" from "run", we can make this script more portable.

In the other copy of the file (see below), we just strip off the "working-directory" piece.
___
*[`drake_cmake_installed/.github/workflows/ci.yml` line 1 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-external-examples/331#-OA-iJNI8gD6BXz7XxSq:-OA-iJNI8gD6BXz7XxSr:b-yosxqi) ([raw file](https://github.com/RobotLocomotion/drake-external-examples/blob/4192d150ec8f3ae71806011ea06c14145745afba/drake_cmake_installed/.github/workflows/ci.yml#L1)):*
> ```YAML
> # SPDX-License-Identifier: MIT-0
> ```

FYI We want every one of the subdir styles to have this `ci.yml` file, showing users how to run GitHub Actions when they copy this into their own repo.
___
*[`drake_cmake_installed/.github/workflows/ci.yml` line 5 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-external-examples/331#-OA-iRNl5UD6ROIVp8hp:-OA-iRNl5UD6ROIVp8hq:br20fq7) ([raw file](https://github.com/RobotLocomotion/drake-external-examples/blob/4192d150ec8f3ae71806011ea06c14145745afba/drake_cmake_installed/.github/workflows/ci.yml#L5)):*
> ```YAML
> ---
> name: ci
> on:
> ```

FYI This top part ("on" and "concurrency") is copied from our top-level `ci.yml`.
___
*[`drake_cmake_installed/.github/workflows/ci.yml` line 19 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-external-examples/331#-OA-iHgBEKTzvd58vg9V:-OA-iHgBEKTzvd58vg9W:b-8v5ux5) ([raw file](https://github.com/RobotLocomotion/drake-external-examples/blob/4192d150ec8f3ae71806011ea06c14145745afba/drake_cmake_installed/.github/workflows/ci.yml#L19)):*
> ```YAML
>   group: ${{ github.head_ref || github.run_id }}
>   cancel-in-progress: true
> jobs:
> ```

FYI This bottom part ("jobs") is copied from our top-level `cmake_installed.yml`.

You (kitware) will need to write a test (either directly in `file_sync_test.py`, or as a sibling to to `file_sync_test.py` and part of the same CI job) that checks if this file is up-to-date with respect to the two places it copied from.

In other words, even though this file is maintained "by hand" in some sense, it's content is always mindless copy-paste from the other, real GHA yml files.

**Comment by jwnimmer-tri** [comment]
Moving to #333 for final reviews & discussions.
